pkg/virtctl/expose/expose.go

```diff
@@ -220,10 +225,36 @@ func (o *Command) RunE(cmd *cobra.Command, args []string) error {
 		service.Spec.ExternalIPs = []string{externalIP}
 	}
 
-	// try to create the service on the cluster
-	_, err = virtClient.CoreV1().Services(namespace).Create(context.Background(), service, metav1.CreateOptions{})
+	major, minor, err := serverVersion(virtClient)
 	if err != nil {
-		return fmt.Errorf("service creation failed: %v", err)
+		return err
+	}
+
+	if major > 1 || (major == 1 && minor >= 20) {
+		_, err = virtClient.CoreV1().Services(namespace).Create(context.Background(), service, k8smetav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("service creation failed for k8s >= 1.20: %v", err)
+		}
+		// For k8s < 1.20 we have to "migrate" the "ipFamilies" field to
+		// "ipFamily" we do this using an unstructured approach
+	} else {
+		// convert the Service to unstructured.Unstructured
+		unstructuredService, err := runtime.DefaultUnstructuredConverter.ToUnstructured(service)
+		if err != nil {
+			return err
+		}
+
+		// Add ipFamily field with proper content
+		err = unstructured.SetNestedField(unstructuredService, string(ipFamily), "spec", "ipFamily")
```

Comment:
should `IPFamilies` be removed ?

Reply:
Let's keep it so we have the original value, k8s is going to ignore it.

Reply:
Ack,
question, does it work other way around, and then you can always have the two fields and remove the convert logic ?